gptqmodel/quantization/config.py

```diff
@@ -283,6 +283,33 @@ class QuantizeConfig():
     # VRAM allocation strategy for MoE-heavy subsets
     vram_strategy: VramStrategy = field(default=VramStrategy.EXCLUSIVE)
 
+    # Control whether to wait for layer finalization (packing, writing) before proceeding to next layer
+    # Default False preserves current behavior (async finalization in background while next layer starts)
+    vram_opt_memory_cleanup_on_stage_end: bool = field(
+        default=False,
+        metadata={"help": "Also wait for all layer finalization tasks (packing, writing) to complete before proceeding to next layer"}
+    )
```

Comment:
rename `vram_opt_memory_cleanup_on_stage_end` to `wait_for_submodule_finalizers`

Small edit
```
metadata={"help": "Wait for all layer finalization tasks (packing, offloading to disk, etc) to complete before proceeding to next layer. May reduce vram pressure for some env."}
```

Reply:
@Qubitium  Remember this also turns off `auto_gc` in favor of `gc` and `torch_empty_cache` on after forward pass of subset and after quantization of subset. Is the `wait_for_submodule_finalizers` name is still appropriate?

Reply:
@avtc  I think we can split this control into two distinct controls?

`wait_for_submodule_finalizers`: bool
`layer_end_memory_gc`: bool

I can cases where I want to enable the first but the second in my own usage. What do you think?

Reply:
done